### ISSUE TITLE
Examples not up to date

### ISSUE THREAD

**Issue by justjacksonn** [mulesoft/raml-for-jax-rs#94]
Maybe I am missing something, but looking at the examples (https://github.com/mulesoft/raml-for-jax-rs/tree/master/raml-to-jaxrs) it seems easy enough.. however two things are happening. First, when I go to return my generated type (I am using JAX-RS 2, not 1.1), I don't see the same methods in the method return type:

```
@Override
    public GetPresentationsResponse getPresentations(final String authorization,
                                                     final String title,
                                                     final Long start,
                                                     final Long pages)
    {
        if (!"s3cr3t".equals(authorization))
        {
            return GetPresentationsResponse.withUnauthorized();
        }
```

the GetPresentationsResponse.withXxx...  is not available to me in any of my generated code. It appears that I get the basic JaxRs Response object type as my class extends ResponseWrapper, so it seems I need to typecase it from Response to my specific response type.

Perhaps we can add some more examples for JAX-RS 2 and maybe some complicated RAML + JSON Schema with complex types (refs, etc) in it to see a more realistic example?

**Comment by ddossot** [comment]
I'm using the RAML->JAX-RS plug-in on this project ( https://github.com/openanalytics/RPooli ), which uses JAX-RS 2 too. I don't need to do any type casting to respond ( for ex: https://github.com/openanalytics/RPooli/blob/master/webapp/src/main/java/eu/openanalytics/rpooli/api/ConfigResource.java#L57 ).

Can you show the bit of RAML you use to specify the `unauthorized` response?

**Comment by justjacksonn** [comment]
Ok..I am going to close this issue. First of all, I can not explain why, but when I created it, the withJson or withUnauthorized, etc were NOT code completing for me.. and they were causing compile issues. Now, it seems to work..and I am not entirely sure what I did that fixed the issue. Maybe some classpath issue that I resolved...   Secondly I didn't have the complete set of jackson/jersey libraries in my classpath so when I deployed it (by doing a cast..which I knew would fail at runtime), it also failed on the reader/writer json classes due to not being in the classpath.  So I am going to chalk this up as a wasted "bug" and wish I could figure out what I did differently so as to offer it as a solution for others that may come across this issue.
